File: from-3301-to-3400/3386-button-with-longest-push-time/solution.go
func buttonWithLongestTime(events [][]int) int {
    maxButton, maxTime := events[0][0], events[0][1]
    prevTime := events[0][1]

    for i := 1; i < len(events); i ++ {

        currButton, currTime := events[i][0], events[i][1]
        elapsedTime := currTime - prevTime
        if elapsedTime > maxTime {
            maxTime = elapsedTime
            maxButton = currButton
        } else if elapsedTime == maxTime && maxButton > currButton {
            maxButton = currButton
        }

        prevTime = currTime
    }
    return maxButton
}

File: from-3301-to-3400/3386-button-with-longest-push-time/solution.py
class Solution:
    def buttonWithLongestTime(self, events: List[List[int]]) -> int:
        max_button, prev_time = events[0]
        max_time = prev_time

        for button, time in events[1:]:
            elapsed_time = time - prev_time
            if elapsed_time > max_time:
                max_time = elapsed_time
                max_button = button
            elif elapsed_time == max_time and max_button > button:
                max_button = button

            prev_time = time

        return max_button